pkg/controller/nodelifecycle/scheduler/timed_workers.go

```diff
@@ -52,11 +52,17 @@ type TimedWorker struct {
 // createWorker creates a TimedWorker that will execute `f` not earlier than `fireAt`.
 func createWorker(ctx context.Context, args *WorkArgs, createdAt time.Time, fireAt time.Time, f func(ctx context.Context, args *WorkArgs) error, clock clock.WithDelayedExecution) *TimedWorker {
 	delay := fireAt.Sub(createdAt)
+	fWrappedWithErrorLogging := func(ctx context.Context, args *WorkArgs) {
+		err := f(ctx, args)
+		if err != nil {
+			klog.Errorf("Timed worker failed with error: %q", err)
+		}
+	}
 	if delay <= 0 {
-		go f(ctx, args)
+		go fWrappedWithErrorLogging(ctx, args)
 		return nil
 	}
-	timer := clock.AfterFunc(delay, func() { f(ctx, args) })
+	timer := clock.AfterFunc(delay, func() { fWrappedWithErrorLogging(ctx, args) })
```

Comment:
this is simpler:

```suggestion
	timer := clock.AfterFunc(delay, func() {
		err := f(ctx, args)
		if err != nil {
			klog.Errorf("Timed worker failed with error: %v", err)
		}
	})
```

Reply:
With the `fWrappedWithErrorLogging` function I want to avoid code duplication as in case `delay<=0` there is another invocation of `f` which was not wrapped with logging too.

Reply:
Ahh, good point